src/backend/distributed/executor/executor.c

```diff
@@ -1576,6 +1592,26 @@ ManageWorkerPool(WorkerPool *workerPool)
 		 * than the target pool size.
 		 */
 		newConnectionCount = Min(newConnectionsForReadyTasks, maxNewConnectionCount);
+
+		if (newConnectionCount > 0 && ExecutorSlowStartInterval > 0)
+		{
+			TimestampTz now = GetCurrentTimestamp();
+
+			if (TimestampDifferenceExceeds(workerPool->lastConnectionOpenTime, now,
+										   ExecutorSlowStartInterval))
+			{
+				newConnectionCount = Min(newConnectionCount,
+										 workerPool->maxNewConnectionsPerCycle);
+
+				/* exponentially increase the open rate (like TCP slow start) */
+				workerPool->maxNewConnectionsPerCycle += 1;
```

Comment:
I'm not sure if I am understanding this correctly. How is this exponential?

Reply:
It isn't, but it was initially. the comment is stale. Will fix it.